pkg/scheduler/scheduler.go

```diff
@@ -282,25 +282,31 @@ func (s *Scheduler) cleanUpAllBindingsFor(ctx context.Context, crp *fleetv1beta1
 		return controller.NewAPIServerError(false, err)
 	}
 
-	// Remove the scheduler cleanup finalizer from all the bindings, and delete them.
-	//
+	// Remove scheduler CRB cleanup finalizer from deleting bindings.
+	for idx := range bindingList.Items {
+		binding := &bindingList.Items[idx]
+		controllerutil.RemoveFinalizer(binding, fleetv1beta1.SchedulerCRBCleanupFinalizer)
+		if err := s.client.Update(ctx, binding); err != nil {
+			klog.ErrorS(err, "Failed to remove scheduler reconcile finalizer from cluster resource binding", "clusterResourceBinding", klog.KObj(binding))
+			return controller.NewUpdateIgnoreConflictError(err)
+		}
+	}
+
 	// Note that once a CRP has been marked for deletion, it will no longer enter the scheduling cycle,
 	// so any cleanup finalizer has to be removed here.
 	//
 	// Also note that for deleted CRPs, derived bindings are deleted right away by the scheduler;
 	// the scheduler no longer marks them as deleting and waits for another controller to actually
 	// run the deletion.
 	for idx := range bindingList.Items {
-		binding := bindingList.Items[idx]
+		binding := &bindingList.Items[idx]
```

Comment:
any reason to change here? Is it this is more performant?

Reply:
Not necessarily, wanted to ensure the update and delete we issued on the same binding using address

Reply:
is there any difference between this and getting the address inside the function call as we normally do, see below?
```
		if err := s.client.Update(ctx, &binding); err != nil {

	       if err := s.client.Delete(ctx, &binding); err != nil && !errors.IsNotFound(err) {
````

Reply:
No difference, But as for performance, It looks like every time we use & in golang we create a new pointer.